carmm/analyse/GCN.py

```diff
@@ -0,0 +1,134 @@
+def coord_number(atoms, lattice='fcc', siteIndices=None):
+    a=3.6
+    # The list that stores coordination number for each atom
+    cn_list = []
+    # The list that stores the indices of first nearst neighbours for each atom
+    # This is a list of fnn index lists
+    fnn_list = []
+
+    if lattice == 'fcc':
+        bond = round(a / 2 ** 0.5, 3)
+    elif lattice == 'bcc':
+        bond = round(a * (3 ** 0.5) / 2, 3)
+
+
+    for i in siteIndices:
+        cn = 0
+        # List for first nearest neighbours of atom i
+        fnn = []
+        # Counting coordination number for atom i
+        for atom_j in atoms:
+            j = atom_j.index
+            # Skip the iteration if we are considering the distance
+            # between atom i and itself
+            if i == j:
+                continue
+            elif abs(atoms[i].z-atom_j.z) > 4:
+                continue
+            # Check if atom i and atom j are first nearst neighbours
+            dist = atoms.get_distance(i, j, mic=True)
+            if dist <= bond+0.001 and dist >= bond-0.001:
+                cn += 1
+                fnn.append(j)
+
+        # Append coordination number and first nearest neighbours to the lists every time the second j loop finishes.
+        cn_list.append(cn)
+        fnn_list.append(fnn)
+
+    return cn_list, fnn_list
+
+def fnn_set(fnnLists):
+    """
+    :param fnnLists
+    :return: Set of all fnns
```

Comment:
These terms aren't clear - needs full expansion of comments to be user friendly. What is fnn?